tests/test_directives.py

```diff
@@ -1735,3 +1735,15 @@ def test_ramsey_example_blog():
     ).strip()
 
     assert full_prog.to_qasm(encal_declarations=True) == expected
+
+
+def test_subroutine_type_hints():
+    @subroutine
+    def test(prog: Program, x) -> IntVar:
```

Comment:
L690 does not test it. The requirement to test is a subroutine with a parameter with a missing type hint. Here, there's `x` with no type hint

Reply:
Oh, I see that `y` is also missing a type hint. Previously, the error was raised by line ` input_ = inputs[argname] = type_hints[argname](name=argname)` rather than by the `ValueError` we throw. Let me verify, and maybe I can update pytest to check the error message

Reply:
Ok! Thank you for noticing this -- I wasn't able to repro with oqpy directly. I think this can be fixed in AutoQASM instead. There's some assumptions in `subroutine` here that aren't respected in AutoQASM